compiler/noirc_evaluator/src/ssa/plonky2_gen/circuit.rs

```diff
@@ -102,6 +122,11 @@ impl Plonky2Circuit {
                 return Err(Plonky2GenError::ICE { message: error_message });
             }
         };
+        if !Self::verify_public_inputs_in_proof(public_inputs, &deserialized_proof) {
+            return Err(Plonky2GenError::ICE {
+                message: "Public inputs don't match proof".to_string(),
+            });
```

Comment:
Please add a new error type, e.g. Plonky2VerifyError to make the error message more informative. Currently, it is
"Error: Backend does not support ICE { message: "Public inputs don't match proof" }.", but could be "Error: Backend does not support VerificationFailed { message: "Public inputs don't match proof" }." as a start. Ideally, we would introduce a new BackendError as well, but perhaps this is outside the scope of this PR.

Reply:
Done: https://github.com/blocksense-network/noir/pull/6/commits/9d8f18cb5605f31300f43217876828f753f285bf